src/components/HathorHeader.js

```diff
@@ -16,59 +16,102 @@ import chevronLeft from '../assets/icons/chevron-left.png';
-      );
-    }
-    return <View style={styles.iconWrapper} />;
-  };
+const HathorHeader = ({
+  title,
+  rightElement,
+  withLogo,
+  withBorder,
+  onBackPress,
+  onCancel,
+  wrapperStyle,
+  children,
+}) => {
+  const hasChildren = children != null;
+  const left = React.Children.toArray(children).find(
+    (child) => child.type.displayName === HathorHeaderLeft.displayName
+  );
+  const right = React.Children.toArray(children).find(
+    (child) => child.type.displayName === HathorHeaderRight.displayName
+  );
 
-  const CancelButton = () => (
-    <SimpleButton
-      icon={closeIcon}
-      onPress={props.onCancel}
-    />
+  return (
+    <Wrapper withBorder={withBorder} style={wrapperStyle}>
+      {hasChildren
+          && (
+          <InnerWrapper>
+            {left}
+            {right}
+          </InnerWrapper>
+          )}
+      {!hasChildren
+          && (
+          <InnerWrapper>
+            <LeftComponent onBackPress={onBackPress} />
+            <CentralComponent title={title} withLogo={withLogo} />
+            <RightComponent rightElement={rightElement} onCancel={onCancel} />
+          </InnerWrapper>
+          )}
+    </Wrapper>
   );
+};
+
+const Wrapper = ({ withBorder, style, children }) => (
+  <View style={[styles.wrapper, style, withBorder && styles.wrapperWithBorder]}>
+    {children}
+  </View>
+);
+
+const InnerWrapper = ({ children }) => (
+  <View style={styles.innerWrapper}>
+    {children}
+  </View>
+);
+
+const HathorHeaderLeft = ({ children }) => (<View>{children}</View>);
+HathorHeaderLeft.displayName = 'HathorHeaderLeft';
+
+const HathorHeaderRight = ({ children }) => <View>{children}</View>;
+HathorHeaderRight.displayName = 'HathorHeaderRight';
+
+HathorHeader.Left = HathorHeaderLeft;
+HathorHeader.Right = HathorHeaderRight;
+
+const CancelButton = ({ onCancel }) => (
+  <SimpleButton
+    icon={closeIcon}
+    onPress={onCancel}
+  />
+);
 
-  const renderHeaderRight = () => {
-    const element = (props.onCancel ? <CancelButton /> : props.rightElement);
+const LeftComponent = ({ onBackPress }) => {
+  if (onBackPress) {
     return (
-      <View style={[styles.iconWrapper, { justifyContent: 'flex-end' }]}>
-        {element}
+      <View style={[styles.iconWrapper, { justifyContent: 'flex-start' }]}>
+        <TouchableOpacity onPress={onBackPress}>
+          <Image source={chevronLeft} width={24} height={24} />
+        </TouchableOpacity>
       </View>
     );
-  };
-
-  const renderHeaderCentral = () => {
-    if (props.withLogo) {
-      return (
-        <Logo
-          style={{ height: 22, width: 100 }}
-        />
-      );
-    }
-    return <Text>{props.title}</Text>;
-  };
+  }
+  return <View style={styles.iconWrapper} />;
+};
 
-  let extraStyle = {};
-  if (props.withBorder) {
-    extraStyle = { borderBottomWidth: 1 };
+const CentralComponent = ({ title, withLogo }) => {
+  if (withLogo) {
+    return (
+      <Logo
+        style={{ height: 22, width: 100 }}
```

Comment:
Move those styles to the stylesheet you created below

Reply:
Done